oaknational/Oak-Web-Application
# [ ENG-49 ] spike curriculum api 2023 depending on url param

## Description

- makes `[viewType]` param in beta path, which can take value:
  - "teachers"
  - "teachers-2023"
- write function `replaceViewType2023`, which:
  - if the user has manually navigated to the `/beta/teachers-2023` version of the site, will keep them in that version by replacing the `/teachers/` part of the link href with `/teachers-2023`
  - if the user is in `/beta/teachers`, will leave link hrefs unchanged, so the user stages in the beta teachers experience.
- adds `/curriculum-api-2023` with a `teachersHomePage.gql` query
- in `/beta/[viewType]/index.ts` (aka teachers homepage):
  - fetch data from curriculum-api or curriculum-api-2023 depending on context.params.viewType (see getStaticProps)

## Issue(s)

Fixes #1713 

Notion ticket: https://www.notion.so/oaknationalacademy/7a067876b8d94be0b4100d6c831d5a1a

## How to test

1. Go to any page in the beta.
2. Change "teachers" in url to "teachers-2023"
3. Site should still work, and clicking around should keep you in "teachers-2023"
4. If you go to teachers homepage (in teachers-2023), you should see different key stages (these are key stages fetched from the new curric api
